src/NativeScript/GlobalObject.moduleLoader.mm

```diff
@@ -390,6 +393,15 @@
                                                   args.append(moduleKey);
                                                   JSValue entry = JSC::call(execState, function, callType, callData, moduleLoader, args);
                                                   record = jsCast<JSModuleRecord*>(entry.get(execState, Identifier::fromString(execState, "module")));
+                          
+                                                  if (frame.check()) {
```

Comment:
If I have understood the API usage correctly, instrumenting a piece of code requires the following changes:
 * Create a frame (`tns::instrumentation::Frame frame;`) just before the code that needs to be instrumented.
 * At the end of the code to be profiled we check if 16000 microseconds have passed since the frame creation (I can't see why this is needed?) and only then we have some logic specific to the event we are profiling, in most cases ending with logging. 

What bothers me is that doing so, we are mixing the profiling-related logic with the actual code and if one day we are profiling a more significant number of events, this may lead to higher maintenance cost. Ideally, we should have a one-liner at the end of the code to be profiled. I understand that the ending logic can't be extracted in a single method because it is different depending on the code/event we are profiling (we are logging different kind of event-specific information), but still we can have event specific methods: `endOfMyCustomEvent1WithFrame::withCustomParam:`, `endOfEvent2withFrame:param2:param3:` etc.